### DISCUSSION TITLE
Error deploying in local

### DISCUSSION THREAD

**Discussion by onmax** [alastria/alastriaID-truffle-contracts#39] [Q&A]
#### How to reproduce it? _Can also see [this gist](https://gist.github.com/onmax/39934d46b5ec49e33ff7521006e794f9)_
- Branch `develop`, with [last commit](https://github.com/alastria/alastriaID-truffle-contracts/commit/79cf65d4caa3aed65fc3c273019b4383bff1f877)
- Using `node v14.16.0`
- Installing dependencies using `npm install`
- (*) Using a local geth server:
```bash
geth --miner.gasprice 0 --http --http.corsdomain="https://remix.ethereum.org" --http.api web3,eth,debug,personal,net --vmdebug --datadir /tmp/geth --dev console
```
- Run in another terminal: `npm run migrateLocal`
_* This problem has also been tested with ganache in a docker_
#### Error
Running the last command will return the following error (geth on the left and deploy on the right):
![photo_2021-07-19_18-22-28](https://user-images.githubusercontent.com/22072217/129768287-3889eb18-5c9c-4194-8566-3552e5489edf.jpg)

#### Workaround
Before, publishing this issue, I've talked with @VictorNS69 which gave me a patch to solve this issue:
- Update dependencies
- Update migrations scripts (5 and 6 have been not fixed)

You can see those changes in this [PR](https://github.com/onmax/alastriaID-truffle-contracts/pull/1) in my own repository

Probably, there is a better way of addressing this issue but we don't know how

**Comment by VictorNS69** [comment]
The problem (_I think_) is that when I try to add these flags `{ gas: 0x6691b7, gasPrice: 0x0 }` to the [`truffle-config.js`](https://github.com/alastria/alastriaID-truffle-contracts/blob/develop/truffle-config.js#L50) file, they do not seem to load and are not set.

Does anyone know if I am doing something wrong, or maybe it is a truffle bug?

We also updated truffle dependencies:
```json
    "@truffle/config": "1.3.1",
    "@truffle/hdwallet-provider": "1.4.2",
    "keythereum": "1.2.0",
    "remixd": "0.2.4-alpha.0",
    "truffle": "5.4.2"
```

Thank you in advance!

**Reply by ghost** [reply]
Victor! I have solved similar issues in the past by setting the gas limit and gas price in the specific transaction that is failing. Should be an additional "options" parameter like Contract.function(param1, param2, {gas: "0x6691b7", gasPrice: "0x0"}). I hope it helps!

**Reply by VictorNS69** [reply]
This is exactly what we did. You can check our changes in this [PR](https://github.com/onmax/alastriaID-truffle-contracts/pull/1/files#diff-f910cef5511c511b2111987a946b14b1c9921aaab007e9cbd9df38cdbdff78dbR4). Our problem is that, we needed to add these flags in almost all the functions, so we wanted to add the flags in a general way for all the truffle project, using the `truffle-config.js` ([networks sections](https://github.com/alastria/alastriaID-truffle-contracts/blob/develop/truffle-config.js#L50)), but it did not work for us.

Our goal is to generalize these flags and get closer to a better solution than adding all flags all the time.

**Comment by onmax** [comment]
I have checked the new README, and it was helpful.

I have seen that you have added a [Ganache section](https://github.com/alastria/alastriaID-truffle-contracts#ganache--truffle) so I had to try it.

## Ganache
I used the command `ganache-cli -g 0x0 -l 0x2FEFD800` directly on your `alastriaID-truffle-contracts` (without the workaround of `{ gas: 0xaf79e0, gasPrice: 0x0 }`) and it worked. I managed to deploy contracts 1 to 4 (`migrateLocal` and `initLocal`).

## geth
Knowing that the new param `-l 0x2FEFD800` is allowing to deploy in ganache, I tried it with geth. I look it up, and it is the gas limit. So, I try it with `geth` and `--miner.gaslimit` as follows:
`geth --miner.gasprice 0 --miner.gaslimit 0x2FEFD800 --http --http.corsdomain="https://remix.ethereum.org" --http.api web3,eth,debug,personal,net --vmdebug --datadir /tmp/geth` 

But it didn't work:
![image](https://user-images.githubusercontent.com/22072217/141508905-160b70e4-d3ff-4ff2-b6fd-7f93f5d785a3.png)

Anyway, I am going to stick with Ganache because it is the one that works
